src/v/raft/tests/foreign_entry_test.cc

```diff
@@ -136,10 +141,13 @@ struct foreign_entry_fixture {
     }
     ~foreign_entry_fixture() {
         _storage.stop().get();
+        _memory_sampling_service.stop().get();
         _feature_table.stop().get();
     }
     model::offset _base_offset{0};
+    ss::logger _test_logger{"foreign-test-logger"};
```

Comment:
Why is the memory sampling service showing up in these apparently unrelated tests?

Does the fixture somehow require it?

Reply:
> I guess this was because of my suggestion to have the reclaim process trigger the LWM logger, right?

Yeah, one of the reasons why I did the timer based thing before was because I wanted to avoid all these changes. Though that's just being a bit lazy really and the notify based one certainly seems better. The reclaim API suggestion would certainly help.